### PR TITLE
rhcos-puddle: start rebuilding openshift-clients too

### PR DESCRIPTION
**Pull request by sosiouxme**
ref. https://jira.coreos.com/browse/ART-844

also move move scheduled-job to proper jobs/build
also, add notification on failure (ref. https://jira.coreos.com/browse/ART-752)

### CODE REVIEW

File: jobs/build/rhcos-puddle/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                    "openshift": {
+                        buildlib.commonlib.shell("./rebuild_rpm.sh openshift ${version}")
+                    }
+                ]
+                if (version != "4.1") {
+                    // in 4.2+ openshift-clients is split out to its own package
+                    builds["clients"] = {
+                        // commonlib.shell isn't safe for concurrency.
+                        // TODO: find the card I know I wrote explaining that.
+                        sh("./rebuild_rpm.sh openshift-clients ${version}")
```
[the remark this answers is not in the document]
Comment (sosiouxme) [reply jobs/build/rhcos-puddle/Jenkinsfile]: I'm super sick of seeing this everywhere. I think Adam tried and failed to find a better way to make this global. If you've got one, let's have it.

Honestly, I'd rather keep it in the script, closer to the bit where you can see why it's required.

File: jobs/build/rhcos-puddle/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            echo "Package build failed:\n${e}"
+            buildException << e
+            currentBuild.result = "FAILURE"
+        }
+
+        stage("run puddle for ${version}") {
+            try {
+                echo "Initializing puddle build #${currentBuild.number} for Openshift version ${version}"
+
+                def puddleConf = "https://raw.githubusercontent.com/openshift/aos-cd-jobs/master/build-scripts/puddle-conf/atomic_openshift-${version}.el8.conf"
```
[the remark this answers is not in the document]
Comment (sosiouxme) [reply jobs/build/rhcos-puddle/Jenkinsfile]: I'd love it actually. In fact, this job was always a bit of a hack; instead it should call out to the signed compose job (always build signed, right?). If I could find a good way to do the RPM rebuild elsewhere (maybe in doozer?) I'd decommission this whole thing in a heartbeat.

File: jobs/build/rhcos-puddle/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                buildException << e
+                currentBuild.result = "FAILURE"
+            }
+        }
+    }
+
+    if(buildException) {
+        err = buildException[0]
+        currentBuild.description = "\nerror: ${err.getMessage()}"
+        commonlib.email(
```
[the remark this answers is not in the document]
Comment (sosiouxme) [reply jobs/build/rhcos-puddle/Jenkinsfile]: :+1:

File: jobs/build/rhcos-puddle/rebuild_rpm.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+tmpdir=$(mktemp -d XXXXXXXXXX.distgit)
+pushd $tmpdir
+
+export REQUESTS_CA_BUNDLE=/etc/pki/tls/certs/ca-bundle.crt
+rhpkg --user ocp-build clone "$package" --branch "rhaos-$version-rhel-8"
+cd "$package"
+git checkout "origin/rhaos-$version-rhel-7" -- .
+rc=0
+if git commit -m "Automated copy from RHEL7 branch"; then
```
[the remark this answers is not in the document]
Comment (sosiouxme) [reply jobs/build/rhcos-puddle/rebuild_rpm.sh]: This isn't really new :)

First it clones the `-rhel-8` branch.
Then it checks out the _content_ (not the commits) from the `-rhel-7` branch.
So it's copying from the rhel7 branch to rhel8 branch.
That said, if you can think of a better commit message, by all means...

**Comment by tbielawa** [comment]
lgtm, looks like you nailed those major points

**Pull request merged**